autopilot/hostfilter.go

```diff
@@ -117,6 +117,9 @@ func isOutOfCollateral(cfg api.AutopilotConfig, s rhpv2.HostSettings, c api.Cont
 // collateral.
 func isBelowCollateralThreshold(cfg api.AutopilotConfig, s rhpv2.HostSettings, c types.Currency) bool {
 	initialCollateral := rhpv2.ContractFormationCollateral(cfg.Contracts.Storage/cfg.Contracts.Amount, cfg.Contracts.Period, s)
+	if initialCollateral.IsZero() {
+		return true // protect against division-by-zero
```

Comment:
Fixes an edge case around hosts which set their `MaxHostCollateral` to 0.

Reply:
This sanity check is fine to leave in but shouldn't we fix this in `ContractFormationCollateral` and only cap if `MaxHostCollateral` is not zero? Do we want to form contracts with hosts that have `MaxHostCollateral` set to 0 in the first place?